src/audio/dai-zephyr.c

```diff
@@ -751,8 +751,17 @@ static int dai_set_dma_config(struct dai_data *dd, struct comp_dev *dev)
 	for (i = 0; i < dma_cfg->block_count; i++) {
 		dma_block_cfg->dest_scatter_en = config->scatter;
 		dma_block_cfg->block_size = config->elem_array.elems[i].size;
-		dma_block_cfg->source_address = config->elem_array.elems[i].src;
-		dma_block_cfg->dest_address = config->elem_array.elems[i].dest;
+		if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
```

Comment:
Why add a new `if` based on direction? 
You can use the one below, from line [765](https://github.com/thesofproject/sof/pull/8948/files#diff-cf0363219538c1e144ce83bdf7ba8aef8c141f26ffe6e2dafff869623360875bR765).

Reply:
my bad, fixed!